addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/enrichment/async/DetectCelebritiesEnrichmentProvider.java

```diff
@@ -84,16 +89,32 @@ public void init(EnrichmentDescriptor descriptor) {
      * Processes the result of the call to AWS
      */
     public Collection<EnrichmentMetadata> processResults(BlobTextFromDocument blobTextFromDoc, String propName,
-            GetCelebrityRecognitionResult result) {
-        List<AIMetadata.Tag> tags = result.getCelebrities()
-                                          .stream()
-                                          .map(c -> newCelebrityTag(c.getCelebrity(), c.getTimestamp()))
-                                          .filter(Objects::nonNull)
-                                          .collect(Collectors.toList());
+            String jobId) {
+        RekognitionService rs = Framework.getService(RekognitionService.class);
+        List<AIMetadata.Tag> tags = new ArrayList<>();
+        List<CelebrityRecognition> nativeCelebrityObjects = new ArrayList<>();
+        GetCelebrityRecognitionResult result = null;
+        do {
+            GetCelebrityRecognitionRequest request = new GetCelebrityRecognitionRequest().withJobId(
+                    jobId).withSortBy(CelebrityRecognitionSortBy.TIMESTAMP);
+
+            if (result != null && result.getNextToken() != null) {
+                request.withNextToken(result.getNextToken());
+            }
+            result = rs.getClient().getCelebrityRecognition(request);
+
+            List<AIMetadata.Tag> currentPageTags = result.getCelebrities()
+                                             .stream()
+                                             .map(c -> newCelebrityTag(c.getCelebrity(), c.getTimestamp()))
+                                             .filter(Objects::nonNull)
+                                             .collect(Collectors.toList());
+
+            tags.addAll(currentPageTags);
+            nativeCelebrityObjects.addAll(result.getCelebrities());
+        } while (result.getNextToken() != null);
```

Comment:
it makes me wonder what is the max size of the outcome list. Should it be limited by user's configuration? Should we consider Set to enforce uniqueness?

Reply:
I thought about it and IMO the notion of max size doesn't really make any sense for a video. A 1h long video will have more results than a 10s one most of the time. Uniqueness should already be enforced for a given timestamp.

Reply:
my concern is that this data might be stored into nuxeo tags if the related feature is enabled. Won't it affect the system and UI if we'll have 1000s of tags?

Reply:
OOTB there is no feature to store video rekognition labels as tags. It's doesn't follow the enrichment pipe logic after the request to rekognition. The way results are handled is hardcoded in the listeners. In this case "save as facet".